compiler/resolution/preFold.cpp

```diff
@@ -883,6 +882,23 @@ Expr* preFold(CallExpr* call) {
         }
       }
     }
+  } else if (call->isPrimitive(PRIM_REDUCE_ASSIGN)) {
```

Comment:
It'd be nice to put a comment here like "Translate PRIM_REDUCE_ASSIGN into a call to accumulateOntoState"